src/options/MapOptions.js

```diff
@@ -654,6 +666,47 @@ class MapOptions extends React.Component {
             }
         }
 
+        // These tiles are ensured for races and may not be replaced
+        const ensuredAnomalies = [];
+
+        if(this.state.ensureRacialAnomalies && this.state.pickRaces) {
+            currentRaces.forEach(race => {
+                let anomalies = []
+                let match = false;
+                // If The Clan of Saar are in the game, ensure we have an asteroid field
+                if(race === 'The Clan of Saar') {
+                    anomalies = useProphecyOfKings ? [...tileData.asteroidFields.concat(tileData.pokAsteroidFields)] : [...tileData.asteroidFields];
+                    match = true;
+                // If The Embers of Muaat are in the game, ensure we have a supernova
+                } else if(race === 'The Embers of Muaat') {
+                    anomalies = useProphecyOfKings ? [...tileData.supernovas.concat(tileData.pokSupernovas)] : [...tileData.supernovas];
+                    match = true;
+                // If The Empyrean are in the game, ensure we have a nebulae
+                } else if(race === 'The Empyrean') {
+                    anomalies = useProphecyOfKings ? [...tileData.nebulae.concat(tileData.pokNebulae)] : [...tileData.nebulae];
+                    match = true;
+                // If The Vuil'Raith Cabal are in the game, ensure we have a gravity rift
+                } else if(race === "The Vuil'Raith Cabal") {
+                    anomalies = useProphecyOfKings ? [...tileData.gravityRifts.concat(tileData.pokGravityRifts)] : [...tileData.gravityRifts];
+                    match = true;
+                }
+                if(match && redsToPlace > 0) {
+                    // Check if the tile is already included
+                    const found = anomalies.some(tile=> newSystems.includes(tile));
+                    // If not we add it
+                    if(!found) {
+                        this.shuffle(anomalies);
+                        newSystems.push(anomalies[0]);
+                        ensuredAnomalies.push(anomalies[0]);
+                        allReds.splice(allReds.indexOf(anomalies[0]), 1);
+                        redsToPlace -= 1;
+                    } else {
+                        ensuredAnomalies.push(newSystems.find(system => anomalies.includes(system)));
+                    }
+                }
+            });
+        }
+
```

Comment:
After merging with your other changes, I am still seeing some of the ensured anomalies getting overwritten with wormholes (see ?settings=T60002350FFT030423).

Reply:
This was actually an issue with the checkbox not being checked after a refresh. I fixed this (and moved the checkbox since it requires picked races anyways) in 61c5fc4.